libpod/container_internal.go

```diff
@@ -2738,3 +2741,327 @@ func (c *Container) update(resources *spec.LinuxResources, restartPolicy *string
+
+	if noHealthCheck {
+		if err := c.removeTransientFiles(context.Background(),
+			c.config.StartupHealthCheckConfig != nil && !c.state.StartupHCPassed,
+			c.state.HCUnitName); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	if c.config.StartupHealthCheckConfig != nil && !c.state.StartupHCPassed && changedStartupItervalTimer {
+		c.state.StartupHCPassed = false
+		c.state.StartupHCSuccessCount = 0
+		c.state.StartupHCFailureCount = 0
+		if err := c.save(); err != nil {
+			return err
+		}
+		c.recreateHealthCheckTimer(context.Background(), true, true)
+	}
+
+	if c.config.HealthCheckConfig != nil && c.state.StartupHCPassed && changedItervalTimer {
+		c.recreateHealthCheckTimer(context.Background(), false, false)
+		return nil
+	}
+
+	if c.config.HealthCheckConfig != nil && c.config.StartupHealthCheckConfig == nil && changedItervalTimer {
+		if err := c.createTimer(c.config.HealthCheckConfig.Interval.String(), false); err != nil {
+			logrus.Errorf("Error recreating container %s healthcheck: %v", c.ID(), err)
+			return nil
+		}
+		if err := c.startTimer(false); err != nil {
+			logrus.Errorf("Error restarting container %s healthcheck timer: %v", c.ID(), err)
+			return nil
+		}
+		return nil
+	}
+	return nil
+}
+
+func (c *Container) updateHealthCheckConfiguration(changedHealthCheckConfig *entities.UpdateHealthCheckConfig) error {
+	if changedHealthCheckConfig == nil {
+		logrus.Debug("New HealthCheck configuration is empty.")
+		return nil
+	}
+
+	oldHealthCheck, oldStartupHealthCheck := c.getCopyOfHealthCheckAndStartupHelathCheck()
+	oldHealthCheckOnFailureAction := c.config.HealthCheckOnFailureAction
+	oldHealthLogDestination := c.config.HealthLogDestination
+	oldHealthMaxLogCount := c.config.HealthMaxLogCount
+	oldHealthMaxLogSize := c.config.HealthMaxLogSize
+
+	err := c.changeGlobalHealthCheckConfiguration(changedHealthCheckConfig)
+	if err != nil {
+		return err
+	}
+
+	changedItervalTimer, err := c.changeHealthCheckConfiguration(changedHealthCheckConfig)
+	if err != nil {
+		return err
+	}
+
+	changedStartupItervalTimer, err := c.changeStartupHealthCheckConfiguration(changedHealthCheckConfig)
+	if err != nil {
+		return err
+	}
+	noHealthCheck := changedHealthCheckConfig.NoHealthCheck == "true"
+
+	if err := c.runtime.state.SafeRewriteContainerConfig(c, "", "", c.config); err != nil {
+		// Assume DB write failed, revert to old resources block
+		c.config.HealthCheckConfig = oldHealthCheck
+		c.config.StartupHealthCheckConfig = oldStartupHealthCheck
+		c.config.HealthCheckOnFailureAction = oldHealthCheckOnFailureAction
+		c.config.HealthLogDestination = oldHealthLogDestination
+		c.config.HealthMaxLogCount = oldHealthMaxLogCount
+		c.config.HealthMaxLogSize = oldHealthMaxLogSize
+		return err
+	}
+
+	err = c.resetHealthCheckTimers(noHealthCheck, changedItervalTimer, changedStartupItervalTimer)
+	if err != nil {
+		return err
+	}
+
+	logrus.Debugf("HealthCheck updated for container %s", c.ID())
+	c.newContainerEvent(events.Update)
```

Comment:
If you're calling this from Update() this is unnecessary, we only want one and it should be in Update itself

Reply:
This comment here still seems to be open, we should not be creating two events